lib/rhea-promise/index.ts

```diff
@@ -0,0 +1,662 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
```

Comment:
@grs, @princjef, @pierreca  - It would be super awesome if you can review this file thoroughly and let me know if you see any issues. It would be nice to create a separate repo named github.com/amqp/rhea-promise and publish a package on npm for it.